packages/drivers/routerlicious-driver/src/documentService.ts

```diff
@@ -57,10 +62,15 @@ export class DocumentService implements api.IDocumentService {
         protected documentId: string,
         protected ordererRestWrapper: RouterliciousOrdererRestWrapper,
         private readonly driverPolicies: IRouterliciousDriverPolicies,
-        private readonly blobCache: ICache<ArrayBufferLike>,
-        private readonly snapshotTreeCache: ICache<ISnapshotTreeVersion>,
         private readonly discoverFluidResolvedUrl: () => Promise<api.IFluidResolvedUrl>,
     ) {
+        // Use the maximum allowed by the policy (IDocumentStorageServicePolicies.maximumCacheDurationMs set below)
+        const snapshotCacheExpiryMs: api.FiveDaysMs = maximumSnapshotCacheDurationMs;
+
+        this.blobCache = new InMemoryCache<ArrayBufferLike>();
+        this.snapshotTreeCache = this.driverPolicies.enableInternalSummaryCaching
```

Comment:
Moving this initialization here rather than in the factory means each DocumentService instance will have its own cache.  I believe this is definitively wrong, since for instance the summarizer and interactive clients would no longer share the cache.

I did it to keep the implementation of the policy in the same place as the setting of the policy object itself, but I'll have to find a different approach.